nixos-module.nix

```diff
@@ -37,12 +37,17 @@ let
           cfg.docker.client.package
         ];
         environment.ARION_PREBUILT = config.settings.out.dockerComposeYaml;
+        environment.DOCKER_HOST = mkIf (cfg.backend == "docker-rootless") "unix:///run/user/1000/docker.sock";
         script = ''
           echo 1>&2 "docker compose file: $ARION_PREBUILT"
           arion --prebuilt-file "$ARION_PREBUILT" up
         '';
       };
-    };
+    in
+      if cfg.backend == "docker-rootless" then
+        { _systemd.user.services."arion-${name}" = service; }
```

Comment:
Yeah, somehow I thought docker and docker-rootless are mutual exclusive. I will change it to support both simultaniously

Reply:
Wait, I am a little bit confused. Would you like to start all services as root services?
I think I would start rootless containers as user services, and normal containers as system services.

Reply:
Running a system service as a (system) user with rootless containers is a valid thing to do. It avoids potential confused deputy problems in the docker daemon.

You wouldn't be able to remove as many capabilities from the systemd unit as a non-containerized solution would, but presumably the rootless container runtime still removes those capabilities.

Reply:
Which kind of "deputy problems" do you mean?

I rather think this could create problems on multi-user setups, because the DOCKER_HOST socket is individual to every user.

But don't get me wrong; it certainly is your thing to decide.